addr/ethereum.go

```diff
@@ -0,0 +1,19 @@
+package addr
+
+import (
+	"fmt"
+	"regexp"
+
+	eth "github.com/ethereum/go-ethereum/common"
+	"github.com/go-errors/errors"
+)
+
+var re = regexp.MustCompile(fmt.Sprintf(`(?:0x)?[0-9a-fA-F]{%d}`, eth.AddressLength*2))
+
+// validateBitcoin validates ethereum address. Return error if address is invalid.
+func validateEthereum(addr string) error {
+	if !eth.IsHexAddress(addr) || !re.MatchString(addr) {
```

Comment:
I think there is some function in the ethereum client to validate the address

Reply:
Didn't find